apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +127,57 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp process_package(package, shipping_method_id) do
+    shipping_method =
```

Comment:
The result of `Enum.find` could be `nil`. We must raise an error in that case.
Overall, there is a need to validate the form-submission. @ashish173 can we assume that the validation is done by the controller?
I'm skeptical of that because that _may_ couple the controller with `snitch_core`... You guys decide.

Reply:
we agreed on putting this inside the controllers for now and have a wrapper with 2 functionalities `common api` & `validations`, TBD later once we have more transitions.